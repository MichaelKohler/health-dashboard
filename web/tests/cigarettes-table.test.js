/* global it, expect, jest */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

import CigaretteTable from '../src/cigarette-table.jsx';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

it('renders correctly', () => {
  const data = [{
    createdAt: '2019-08-07T08:00:03Z',
    rolled: false,
  }];

  const wrapper = shallow(<CigaretteTable rows={ data }/>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
