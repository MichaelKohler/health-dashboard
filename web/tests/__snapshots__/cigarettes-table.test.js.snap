// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <table
    className="MuiTable-root makeStyles-table-2"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-head"
          scope="col"
        >
          Date
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-head"
          scope="col"
        >
          Rolled
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-head"
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-body"
        >
          2019-08-07T08:00:03Z
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-body"
        >
          not rolled
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-body"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Delete
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
